std/datetime.d

```diff
@@ -29161,6 +29191,72 @@ private:
         _hasDST = hasDST;
     }
 
+    // Android concatenates the usual timezone directories into a single file,
+    // tzdata, along with an index to jump to each timezone's offset.  In older
+    // versions of Android, the index was stored in a separate file, zoneinfo.idx,
+    // whereas now it's stored at the beginning of tzdata.
+    version(Android)
+    {
+        // Keep track of whether there's a separate index, zoneinfo.idx
+        static bool separate_index;
```

Comment:
If I understand it correctly, this code is run when looking up a timezone by name for the first time in a thread. In terms of space, it allocates one 40 byte chunk from the GC per timezone entry, plus whatever space overhead the hash table needs, which then both stick around for as long as the thread lives. On my Arch Linux system `timedatectl list-timezones | wc -l` is 417.

Programs that multiplex tasks onto a thread pool probably won't be in bad shape, as they tend to use a low number of system threads AFAIU. I guess the worst case is a system that uses a lot of long-lived system threads and they all look up timezones by name at some point. Doesn't sound like a good design, but maybe it's something a user would do anyway.

In terms of time, many of the short-lived GC allocations can be eliminated by allocating on the stack, and many of the file reads can be consolidated into one, but since the result is cached I don't think this is particularly important.

I guess this strategy is probably not a problem in practice. Maybe when std.datetime isn't a horrific monolith of a module, using a module constructor becomes more appealing…